File: benchmarks/pairs/vibe-py/djangoat-analytics-hits-by-ip/fixed.py
# Provenance: Contrast-Security-OSS/DjanGoat hits_by_ip (fixed).
# repo: Contrast-Security-OSS/DjanGoat
# commit: 082c88bfefced508ae083b92b459c808f95e905e
# parent: 082c88bfefced508ae083b92b459c808f95e905e
# commit_url: https://github.com/Contrast-Security-OSS/DjanGoat/blob/082c88bfefced508ae083b92b459c808f95e905e/app/models/Analytics/analytics.py#L57
# cve: 
# license: MIT
# function: hits_by_ip
# relpath: app/models/Analytics/analytics.py
# provenance: human
# mechanism: source_reaches_sink

    def parse_field(field):
        valid_fields = ["ip_address", "referrer", "user_agent"]
        if field in valid_fields:
            return field
        else:
            return '1'

File: benchmarks/pairs/vfc-js/diskusage-ng-0-2-6/fixed.js
// Provenance: iximiuz/node-diskusage-ng  (fixed).
// repo: iximiuz/node-diskusage-ng
// commit: 48e7e093486b528f0c81ec699573e0e4a431b8d3
// parent: 53c505506553c83861d1bd67823d6a382d3a5a43
// commit_url: https://github.com/iximiuz/node-diskusage-ng/commit/48e7e093486b528f0c81ec699573e0e4a431b8d3
// cve: CVE-2020-7631
// license: MIT
// function: diskusage
// relpath: lib/posix.js
// provenance: human
// mechanism: source_reaches_sink
// upstream_start: 6

function diskusage(path, cb) {
    execFile('df', ['-k', path], function(err, stdout) {
        if (err) {
            return cb(err);
        }

        try {
            cb(null, parse(stdout));
        } catch (e) {
            cb(e);
        }
    });
}
